### DISCUSSION TITLE
ThrowOnError param when using Batch

### DISCUSSION THREAD

**Discussion by aramB** [pnp/pnpcore#858] [Q&A] [answered]
Hi,

I have discovered that if a batch contains only 1 (one) request then it will throw exception when executing the batch EVEN IF `throwOnError` param is set to false. Is this some expected behaviour? I mean is it per design or is it a bug?

Thanks in advance!
// Aram

**Comment by jansenbe** [comment, marked as answer]
@aramB : that might be a bug when you're explicitly using a batch method. Can you tell us which batch method you we're using? Or share a little snippet of your code.

**Reply by aramB** [reply]
@jansenbe thank you for your response. The batch method we are using in this case is `EnsureUserBatchAsync`. Here comes a little code example to reproduce the issue (if it is an issue :) ):
```
    var batch = pnpContext.NewBatch();
    var user1 = await pnpContext.Web.EnsureUserBatchAsync(batch, "nonexistinguser1@example.com");
    //var user2 = await pnpContext.Web.EnsureUserBatchAsync(batch, "nonexistinguser2@example.com");
    var results = await pnpContext.ExecuteAsync(batch, throwOnError: false);
    foreach (var result in results)
    {
        Console.WriteLine(result.ApiResponse);
    }
```
By uncommenting `user2` the will not throw any exception.

Another interesting thing that we noticed here is when we uncomment `user2`, the `results` (batch result) contains 4 results instead of 2.

**Reply by jansenbe** [reply]
@aramB : we did not correctly identify all batch requests as batch requests and as such sometimes the call was executed interactively. This just works, but downside is that the batching features (like prevent throw on error) did not work. A fix has been committed that correctly identifies requests as coming from a batch method and as such the throw on error is now respected. This fix will be part of the next nightly build.

Regarding the double entries, this might be another issue...will be looking into that as well.

**Reply by jansenbe** [reply]
@aramB : double entry issue is also fixed, please test with next nightly. 

Thanks for using PnP Core SDK and providing the feedback 💪

**Reply by aramB** [reply]
Awesome! tnx @jansenbe! Any idea when v1.7 will be released?

**Reply by jansenbe** [reply]
Somewhere in June is the current thinking
